### PR TITLE
feat: edit documentation for Account Abstraction and add Smart Backend Wallets

### PR DESCRIPTION
**Pull request by d4mr**
closes INFRA-411

---

## PR-Codex overview
This PR focuses on enhancing the documentation for `Account Abstraction` and `Backend Wallets` in the thirdweb Engine, introducing `Smart Backend Wallets` and their benefits, along with configuration types and pricing details.

### Detailed summary
- Added a `Callout` in `account-abstraction/page.mdx` recommending `Smart Backend Wallets` for backend operations.
- Introduced a new section on `Smart Backend Wallets` in `backend-wallets/page.mdx`.
- Listed benefits of `Smart Backend Wallets`, including gas management and simple integration.
- Explained how `Smart Backend Wallets` differ from previous implementations.
- Provided details on when not to use `Smart Backend Wallets`.
- Specified configuration types for smart accounts.
- Clarified pricing and deployment costs for `Smart Backend Wallets`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by d4mr** [comment]
### Merge activity

* **Nov 25, 4:57 PM EST**: The merge label 'merge-queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=thirdweb-dev&repo=js) once it meets the requirements.

File: apps/portal/src/app/engine/features/backend-wallets/page.mdx [hunk 1]
```diff
@@ -12,6 +13,50 @@ Engine performs blockchain actions using backend wallets that you own and manage
...
+
+<Details summary="How are Smart Backend Wallets different from existing engine AA features?">
+	Unlike the [previous account abstraction implementation in
+	Engine](account-abstraction) where you had to manage both the account address
+	(`x-account-address`) and the backend wallet address
+	(`x-backend-wallet-address`) separately, smart backend wallets simplify this.
+	The backend wallet address is now the smart account address itself.
+</Details>
+
+<Details summary="When not to use Smart Backend Wallets?">
```
Comment (arcoraven) [inline apps/portal/src/app/engine/features/backend-wallets/page.mdx:40]: Hm would someone need this, or would we build this in the future? Maybe this just belongs in an FAQ section at the bottom:

```markdown
## FAQ

#### Can I import a smart account?
No, importing a smart account is currently not supported.
```

File: apps/portal/src/app/engine/features/backend-wallets/page.mdx [hunk 2]
```diff
@@ -12,6 +13,50 @@ Engine performs blockchain actions using backend wallets that you own and manage
...
+	At this time, Smart Backend Wallets do not allow for importing a smart
+	account. In use cases where you need to import a smart account (such as with
+	session keys), you should use [Engine AA features that utilise the
+	`x-account-address` header](account-abstraction).
+</Details>
+
+### Configuration Types
+
+- `smart:local` - Smart account backed by a local key
```
Comment (arcoraven) [inline apps/portal/src/app/engine/features/backend-wallets/page.mdx:49]: "backed up locally in Engine" or something. Don't need to reference the EOA

File: apps/portal/src/app/engine/features/backend-wallets/page.mdx [hunk 3]
```diff
@@ -12,6 +13,50 @@ Engine performs blockchain actions using backend wallets that you own and manage
...
+- `smart:aws-kms` - Smart account backed by AWS KMS
+- `smart:gcp-kms` - Smart account backed by Google Cloud KMS
+
+For AWS and Google Cloud KMS options, follow the setup instructions in the respective sections below.
+
+### Pricing
+
+Smart backend wallets have no additional costs to use with your Engine instance. Transactions sent from smart backend wallets follow the [same billing model as regular account abstraction transactions with thirdweb](/connect/account-abstraction/infrastructure#pricing--billing).
+Smart accounts need to be deployed on each chain. This costs gas and is billed to your account like any other transaction.
```
Comment (arcoraven) [inline apps/portal/src/app/engine/features/backend-wallets/page.mdx:58]: A little more clear:
> Note: Smart Backend Wallets deploy a smart account on each chain they send transactions on. You are billed for this deployment cost.

File: apps/portal/src/app/engine/features/backend-wallets/page.mdx [hunk 4]
```diff
@@ -12,6 +13,50 @@ Engine performs blockchain actions using backend wallets that you own and manage
...
+	`x-account-address` header](account-abstraction).
+</Details>
+
+### Configuration Types
+
+- `smart:local` - Smart account backed by a local key
+- `smart:aws-kms` - Smart account backed by AWS KMS
+- `smart:gcp-kms` - Smart account backed by Google Cloud KMS
+
+For AWS and Google Cloud KMS options, follow the setup instructions in the respective sections below.
```
Comment (arcoraven) [inline apps/portal/src/app/engine/features/backend-wallets/page.mdx:53]: > To configure AWS and Google Cloud KMS, ...

File: apps/portal/src/app/engine/features/backend-wallets/page.mdx [hunk 5]
```diff
@@ -12,6 +13,50 @@ Engine performs blockchain actions using backend wallets that you own and manage
...
+
+### Pricing
+
+Smart backend wallets have no additional costs to use with your Engine instance. Transactions sent from smart backend wallets follow the [same billing model as regular account abstraction transactions with thirdweb](/connect/account-abstraction/infrastructure#pricing--billing).
```
Comment (arcoraven) [inline apps/portal/src/app/engine/features/backend-wallets/page.mdx:57]: We should note the customer is charged for _mainnet_ usage only.

> You are billed for mainnet transactions sent from Smart Backend Wallets (see [Account Abstraction pricing](LINK)). There is no cost for testnet transactions.

**Review by arcoraven — APPROVED** [review]

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**